joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -1785,24 +1791,25 @@ class AstCreator(filename: String, javaParserAst: CompilationUnit, global: Globa
   }
 
   def astForArrayCreationExpr(expr: ArrayCreationExpr, expectedType: Option[ExpectedType]): Ast = {
-    val typeFullName = expressionReturnTypeFullName(expr).orElse(expectedType.map(_.fullName))
-    val callNode     = operatorCallNode(Operators.alloc, code = expr.toString, typeFullName = typeFullName)
+    val maybeInitializerAst = expr.getInitializer.toScala.map(astForArrayInitializerExpr(_, expectedType))
 
-    val levelAsts = expr.getLevels.asScala.flatMap { lvl =>
-      lvl.getDimension.toScala match {
-        case Some(dimension) => astsForExpression(dimension, Some(ExpectedType.Int))
-
-        case None => Seq.empty
-      }
-    }
+    maybeInitializerAst match {
+      case Some(initializerAst) =>
+        initializerAst.root.collect { case call: NewCall => call.code(expr.toString) }
```

Comment:
does this line create a collection and then throw it away for side-effects, or is it a left-over from debugging?

Reply:
oh, it sets the code of the call to `expr.toString` doesn't it? Sometimes you just have to ask a question to find the answer...

Reply:
That is indeed what it does! The fact that you had to ask means I should probably clean it up a bit :)